lib/SILGen/ArgumentSource.cpp

```diff
@@ -217,3 +217,17 @@ void ArgumentSource::forwardInto(SILGenFunction &SGF,
   auto substLoweredType = destTL.getLoweredType().getSwiftRValueType();
   RValue(SGF, loc, substLoweredType, outputValue).forwardInto(SGF, loc, dest);
 }
+
+SILType ArgumentSource::getSILSubstRValueType(SILGenFunction &SGF) const & {
+  CanSILFunctionType funcType = SGF.F.getLoweredFunctionType();
+  CanType substType = getSubstType();
+  AbstractionPattern origType(funcType->getGenericSignature(), substType);
```

Comment:
Is the argument value always emitted at the most substituted abstraction level?

I don't think you need the abstraction pattern at all then. substType should not contain any interface type parameters, so just calling getLoweredType(substType) should work?